pkg/build/api/v1beta3/types.go

```diff
@@ -167,6 +172,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStream tag or a DockerImage to
+	// copy source from
+	From kapi.ObjectReference `json:"from" description:"image reference"`
+
+	// Paths is a list of source and destination paths to copy from the image
+	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
+
+	// PullSecret is a reference to a secret to be used to pull the image from
+	// a registry
+	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"pull secret for source image"`
```

Comment:
Hrm - I still agree with Jordan's point about proving that source secret is
distinct.  If it is, then source secret shouldn't be where it is.

On Dec 15, 2015, at 3:03 PM, Cesar Wong <notifications@github.com> wrote:

In pkg/build/api/v1beta3/types.go
<https://github.com/openshift/origin/pull/6324#discussion_r47689900>:

> @@ -167,6 +172,31 @@ type BuildSource struct {
>  	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
>  }
>
> +// ImageSource describes an image that is used as source for the build
> +type ImageSource struct {
> +	// From is a reference to an ImageStream tag or a DockerImage to
> +	// copy source from
> +	From kapi.ObjectReference `json:"from" description:"image reference"`
> +
> +	// Paths is a list of source and destination paths to copy from the image
> +	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
> +
> +	// PullSecret is a reference to a secret to be used to pull the image from
> +	// a registry
> +	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"pull secret for source image"`

We want a dockercfg here for which we'll normally use the one associated
with the builder service account when pulling from the openshift registry.
If you're pulling from the openshift registry, we'll fill in the secret for
you.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/6324/files#r47689900>.